pkg/controller/nodeipam/ipam/cidrset/cidr_set_test.go

```diff
@@ -700,34 +700,39 @@ func TestCIDRSetv6(t *testing.T) {
 		},
 	}
 	for _, tc := range cases {
-		_, clusterCIDR, _ := net.ParseCIDR(tc.clusterCIDRStr)
-		a, err := NewCIDRSet(clusterCIDR, tc.subNetMaskSize)
-		if err != nil {
-			if tc.expectErr {
-				continue
-			}
-			t.Fatalf("Error allocating CIDRSet for %v", tc.description)
-		}
-
-		p, err := a.AllocateNext()
-		if err == nil && tc.expectErr {
-			t.Errorf("expected error but got none for %v", tc.description)
-			continue
-		}
-		if err != nil && !tc.expectErr {
-			t.Errorf("unexpected error: %v for %v", err, tc.description)
-			continue
-		}
-		if !tc.expectErr {
-			if p.String() != tc.expectedCIDR {
-				t.Fatalf("unexpected allocated cidr: %s for %v", p.String(), tc.description)
+		t.Run(tc.description, func(t *testing.T) {
+			_, clusterCIDR, _ := net.ParseCIDR(tc.clusterCIDRStr)
+			a, err := NewCIDRSet(clusterCIDR, tc.subNetMaskSize)
+			if (a == nil || err != nil) && !tc.expectErr {
```

Comment:
```
if gotErr := err != nil; gotErr != tc.expectErr {
  t.Fatalf("NewCIDRSet(clusterCIDR, tc.subNetMaskSize) = %v, %v; gotErr = %t, want %t", a, err, gotErr, tc.expectErr)
}
if a == nil {
  return
}
...

```

Reply:
thanks, just a few small cleanups to reduce the if- complexity

Reply:
ya sure I will do